src/lib.rs

```diff
@@ -395,6 +395,83 @@ impl<T: Ord> OrderedCollection<T> {
         // SAFETY: 1 <= i, so not [0], so initialized
         (i > 0).then(|| unsafe { self.items.get_unchecked(i).assume_init_ref() })
     }
+
+    /// Iterator over elements of a collection.
+    ///
+    /// It yields all items in an unspecified order.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ordsearch::OrderedCollection;
+    /// let expected = vec![1, 2, 3, 4, 5];
+    /// let coll = OrderedCollection::from(expected.clone());
+    /// let mut values: Vec<_> = coll.iter().copied().collect();
+    /// values.sort();
+    /// assert_eq!(values, expected);
+    /// ```
+    pub fn iter(&self) -> impl Iterator<Item = &T> {
```

Comment:
Let's also `impl IntoIterator for &OrderedCollection`

Reply:
✅